src/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -243,23 +245,31 @@ private static void ConvertMultiLineToSingleLineSpans(SourceText text, Segmented
 
             var tokenTypeMap = SemanticTokensSchema.GetSchema(supportsVisualStudioExtensions).TokenTypeMap;
 
-            using var _ = ArrayBuilder<int>.GetInstance(5 * classifiedSpans.Count, out var data);
-            for (var currentClassifiedSpanIndex = 0; currentClassifiedSpanIndex < classifiedSpans.Count; currentClassifiedSpanIndex++)
+            var data = s_tokenListPool.Allocate();
+            try
             {
-                currentClassifiedSpanIndex = ComputeNextToken(
-                    lines, ref lastLineNumber, ref lastStartCharacter, classifiedSpans,
-                    currentClassifiedSpanIndex, tokenTypeMap, tokenTypesToIndex,
-                    out var deltaLine, out var startCharacterDelta, out var tokenLength,
-                    out var tokenType, out var tokenModifiers);
-
-                data.Add(deltaLine);
-                data.Add(startCharacterDelta);
-                data.Add(tokenLength);
-                data.Add(tokenType);
-                data.Add(tokenModifiers);
-            }
+                for (var currentClassifiedSpanIndex = 0; currentClassifiedSpanIndex < classifiedSpans.Count; currentClassifiedSpanIndex++)
+                {
+                    currentClassifiedSpanIndex = ComputeNextToken(
+                        lines, ref lastLineNumber, ref lastStartCharacter, classifiedSpans,
+                        currentClassifiedSpanIndex, tokenTypeMap, tokenTypesToIndex,
+                        out var deltaLine, out var startCharacterDelta, out var tokenLength,
+                        out var tokenType, out var tokenModifiers);
+
+                    data.Add(deltaLine);
+                    data.Add(startCharacterDelta);
+                    data.Add(tokenLength);
+                    data.Add(tokenType);
+                    data.Add(tokenModifiers);
+                }
 
-            return data.ToArray();
+                return data.ToArray();
+            }
+            finally
+            {
+                data.Clear();
```

Comment:
i am fine not clearing an list of ints.

Reply:
Sure, went ahead and switched to GetPooledObject. Clearing will now occur upon retrieval from the pool.